commands/init.go

```diff
@@ -27,70 +41,174 @@ var initCmd = &cmds.Command{
 		cmdkit.BoolOption(DevnetUser, "when set, populates config bootstrap addrs with the dns multiaddrs of the user devnet and other user devnet specific bootstrap parameters"),
 	},
 	Run: func(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment) error {
-		repoDir := getRepoDir(req)
+		newConfig, err := getConfigFromOptions(req.Options)
+		if err != nil {
+			return err
+		}
+
+		repoDir, _ := req.Options[OptionRepoDir].(string)
 		if err := re.Emit(fmt.Sprintf("initializing filecoin node at %s\n", repoDir)); err != nil {
 			return err
 		}
+		rep, err := repo.CreateRepo(repoDir, newConfig)
 
-		genesisFile, _ := req.Options[GenesisFile].(string)
-		peerKeyFile, _ := req.Options[PeerKeyFile].(string)
-		autoSealIntervalSeconds, _ := req.Options[AutoSealIntervalSeconds].(uint)
-		devnetTest, _ := req.Options[DevnetTest].(bool)
-		devnetNightly, _ := req.Options[DevnetNightly].(bool)
-		devnetUser, _ := req.Options[DevnetUser].(bool)
-
-		var withMiner address.Address
-		if m, ok := req.Options[WithMiner].(string); ok {
-			var err error
-			withMiner, err = address.NewFromString(m)
-			if err != nil {
-				return err
-			}
+		defer func() {
+			if closeErr := rep.Close(); closeErr != nil {
+				if err == nil {
+					err = closeErr
+				} else {
+					err = errors.Wrap(err, closeErr.Error())
+				}
+			} // else err may be set and returned as normal
+		}()
```

Comment:
I don't think this error is going to be handled correctly. I believe the function needs to have a named return value.

Reply:
This was copy pasted from the previous implementation. It turns out we don't need the error or a go func at all. The only condition under which `Close()` throws an error is if the file was already closed. I'm replacing this with a simple defer.